events/2022-google-summer-of-code/index.markdown

```diff
@@ -0,0 +1,49 @@
+---
+author: Mark Moll
+comments: false
+date: 2022-02-17 00:00:00 -0700
+layout: page
+slug: 2022-google-summer-of-code
+title: 2022 Google Summer of Code
+---
+# 2022 Google Summer of Code
+
+MoveIt is planning to participate again in the [Google Summer of Code](https://summerofcode.withgoogle.com). Below is a tentative list of project ideas. If you have your own project idea, feel free to create a pull request on the [repository for this page](https://github.com/ros-planning/moveit.ros.org). Unlike previous years, there are now two different project sizes. Medium projects are 175 hours, while large projects are 350 hours. See the [Google Summer of Code FAQ](https://developers.google.com/open-source/gsoc/faq) for details.
+
+## Project ideas
+
+### Bullet collision checking integration
+
+* Bullet support is partially done, but more work is needed. Support for continuous collision checking would be nice.
+* Project size: medium
+* Mentors: ???
+
+### Python bindings for MoveIt 2
+
+* Use [pybind11](https://pybind11.readthedocs.io/en/latest) to create python bindings for a subset of the MoveIt C++ API.
+* Project size: large
+* Mentors: Henning Kayser
+
+### Improved benchmarking Support in MoveIt
+
+* Continue the work from last year's Google Summer of Code project to create more benchmarking capabilities.
+* Project size: large
+* Mentors: Michael Görner
+
+### Adopt modern C++ best practices
+
+* Help overhaul MoveIt's code to adopt modern C++ best practices. Many examples for doing this can be found in the [Moveit 2 issues that are marked "good first issue."](https://github.com/ros-planning/moveit2/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22).
+* Project size: medium--large
+* Mentor: Tyler Weaver
+
+### Add ability to attach/detach end-effector links
+
+* Add functionality to MoveIt to programmatically attach and detach grippers from a manipulator. See [this issue](https://github.com/ros-planning/moveit2/issues/208) for a more detailed discussion.
```

Comment:
I'm not sure this is actionable enough already? Do we want the student to also solve the design problem? I'm currently implementing this as a hacky add-on were collisions are enabled/disabled and links in the rviz display are shown/hidden but this could be implemented a bit more thoroughly with `RobotState` and `RobotStateMsg` containing the info about disabled parts of the URDF .... or you go for some magical scene graph

Reply:
Simon if you want to mentor this, that's fine with me. Sounds like you are a lot more familiar with the problem than I am

Reply:
There were usually two mentors in the past ... 

But my point was more on that we should propably decide beforehand which route we actually want to take. For my needs the approach I sketched would be totally sufficient but others have proposed more generic solutions

Reply:
@JafarAbdi has a hacky solution for this we used for one client that sounds really similar to what @simonschmeisser you suggested would be the easy solution.

I do think for this to be a successful project we would need to do the design work on how we'd like this done before asking a student to do it.

Reply:
I agree with @tylerjw here. If you want to have this as a project, you should define the approach before the project starts. In the best case together with a group of maintainers with different ideas about it.